2017-02-20  Quentin Minster <quentin@minster.io>

	#https://bugs.gentoo.org/show_bug.cgi?id=591582
	* Config.kmk: add paxmark befor VBOX_VBOXTPG get run.
	* src/VBox/VMM/Makefile.kmk: add paxmark befor VBOX_VBOXCPP get run.
	* src/VBox/Main/Makefile.kmk: add paxmark befor USBIdDatabaseGenerator_1_TARGET get run.

--- a/Config.kmk
+++ b/Config.kmk	2016-09-21 02:22:28 
@@ -3719,11 +3719,13 @@
 		$(if-expr $(intersects $(KBUILD_TARGET_ARCH),$(KBUILD_ARCHES_64)),-64,-32) \
 		-h --host-$(VBOX_HC_ARCH_BITS)-bit #-vvv
 define TOOL_VBoxTpG_DTRACE_HDR_CMDS
+	$(QUIET)paxmark.sh -m "$(VBOX_VBOXTPG)"
 	$(QUIET)$(VBOX_VBOXTPG) $(flags) -o "$(out)" -s "$(source)"
 endef
 TOOL_VBoxTpG_DTRACE_OBJ_NOT_NEEDED :=
 TOOL_VBoxTpG_DTRACE_OBJ_FLAGS      := -G --host-$(VBOX_HC_ARCH_BITS)-bit #-vvv
 define TOOL_VBoxTpG_DTRACE_OBJ_CMDS
+	$(QUIET)paxmark.sh -m "$(VBOX_VBOXTPG)"
 	$(QUIET)$(VBOX_VBOXTPG) \
 		$(if-expr $(intersects $(bld_trg_arch),$(KBUILD_ARCHES_64)),-64,-32) \
 		$(flags) \
@@ -3738,6 +3740,7 @@
 TOOL_VBoxTpG-Disabled_DTRACE_DEPORD     = $(VBOX_VBOXTPG)
 TOOL_VBoxTpG-Disabled_DTRACE_HDR_FLAGS := -h
 define TOOL_VBoxTpG-Disabled_DTRACE_HDR_CMDS
+	$(QUIET)paxmark.sh -m "$(VBOX_VBOXTPG)"
 	$(QUIET)$(VBOX_VBOXTPG) $(flags) -o "$(out)" -s "$(source)"
 endef
 TOOL_VBoxTpG-Disabled_DTRACE_OBJ_NOT_NEEDED := $(KBUILD_OSES)
@@ -3752,6 +3755,7 @@
 TOOL_DTraceAndVBoxTpG_DTRACE_HDR_CMDS =
 define TOOL_DTraceAndVBoxTpG_DTRACE_HDR_CMDS
 	$(TOOL_StandardDTrace_DTRACE_HDR_CMDS)
+	$(QUIET)paxmark.sh -m "$(VBOX_VBOXTPG)"
 	$(QUIET)$(VBOX_VBOXTPG) --generate-wrapper-header --host-$(VBOX_HC_ARCH_BITS)-bit --ring-3-context \
 		$(filter-out -C, $(filter-out -h,$(flags))) \
 		-o "$(out).tmp" -s "$(source)"
--- a/src/VBox/VMM/Makefile.kmk	2018-04-10 15:26:24.622867735 +0300
+++ b/src/VBox/VMM/Makefile.kmk	2018-04-10 15:24:35.901998796 +0300
@@ -896,6 +896,7 @@
   $$(VMMLibDTrace_0_OUTDIR)/$1: $2 $$(VBOX_VBOXCPP) | $$$$(dir $$$$@)
 	$$(call KB_FN_AUTO_CMD_DEPS_COMMANDS)
 	$$(QUIET)$$(call MSG_GENERATE,VMMLibDTrace,$$@,$2)
+	$$(QUIET)paxmark.sh -m "$(VBOX_VBOXCPP)"
 	$$(QUIET)$(VBOX_VBOXCPP) -d \
 		-D VBOX_FOR_DTRACE_LIB \
 		-D VBOX_FOR_DTRACE_LIB_$(toupper $(KBUILD_TARGET_ARCH)) \
--- a/src/VBox/Main/Makefile.kmk	2018-04-10 15:26:15.292793067 +0300
+++ b/src/VBox/Main/Makefile.kmk	2018-04-10 15:25:14.042303341 +0300
@@ -607,6 +607,7 @@
  		$(PATH_SUB_CURRENT)/src-server/usb.ids $(PATH_SUB_CURRENT)/src-server/custom.ids | $$(dir $$@)
 	$(call KB_FN_AUTO_CMD_DEPS_COMMANDS)
 	$(call MSG_GENERATE,USBIdDatabase,$@,$(USBIdDatabaseGenerator_1_TARGET))
+	$(QUIET)paxmark.sh -m "$(USBIdDatabaseGenerator_1_TARGET)"
 	$(QUIET)$(USBIdDatabaseGenerator_1_TARGET) -o "$@" $(filter %.ids,$^)
 
  BLDPROGS += USBIdDatabaseGenerator
